pkg/metrics/server.go

```diff
@@ -0,0 +1,113 @@
+package metrics
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"k8s.io/apiserver/pkg/server"
+	"k8s.io/client-go/tools/clientcmd"
+	basemetrics "k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+
+	buildv1client "github.com/openshift/client-go/build/clientset/versioned"
+	buildv1informers "github.com/openshift/client-go/build/informers/externalversions"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	configv1informers "github.com/openshift/client-go/config/informers/externalversions"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configmetrics"
+	buildmetrics "github.com/openshift/openshift-controller-manager/pkg/build/metrics/prometheus"
+)
+
+type MetricsServer struct {
+	ListenAddress string
+	CertFile      string
+	KeyFile       string
+}
+
+func (s *MetricsServer) Run() error {
+	rules := clientcmd.NewDefaultClientConfigLoadingRules()
+	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(rules, &clientcmd.ConfigOverrides{})
+	cfg, err := kubeconfig.ClientConfig()
+	if err != nil {
+		klog.Errorf("Cannot get cluster configuration: %v", err)
+		return err
+	}
+
+	configClient, err := configv1client.NewForConfig(cfg)
+	if err != nil {
+		klog.Errorf("Cannot get config client: %v", err)
+		return err
+	}
+
+	buildClient, err := buildv1client.NewForConfig(cfg)
+	if err != nil {
+		klog.Errorf("Cannot get build client: %v", err)
+		return err
+	}
+
+	configInformers := configv1informers.NewSharedInformerFactory(configClient, 30*time.Minute)
```

Comment:
did you mean to leave the time as 30 minutes hard coded (just a question no need to change if this is meant to be this way)?

Reply:
Yes, I don't think making this configurable would add much.